src/browserlib/extract-dfns.mjs

```diff
@@ -216,12 +216,36 @@ function definitionMapper(el, idToHeading, usesDfnDataModel) {
 export default function (spec, idToHeading = {}) {
   const definitionsSelector = [
     // re data-lt, see https://github.com/w3c/reffy/issues/336#issuecomment-650339747
+    // As for `<dfn>` we'll consider that headings without a `data-dfn-type`
+    // have an implicit `"data-dfn-type"="dfn"` attribute, provided they also
+    // have some other definition related attribute (because we only want to
+    // extract headings that want to be seen as definitions)
     'dfn[id]:not([data-lt=""])',
     'h2[id][data-dfn-type]:not([data-lt=""])',
     'h3[id][data-dfn-type]:not([data-lt=""])',
     'h4[id][data-dfn-type]:not([data-lt=""])',
     'h5[id][data-dfn-type]:not([data-lt=""])',
-    'h6[id][data-dfn-type]:not([data-lt=""])'
+    'h6[id][data-dfn-type]:not([data-lt=""])',
```

Comment:
I think it might be time to replace this long list with `:is(h2,h3,h4,h5,h6)[data-dfn-type]:not(data-lt=""])` (and re-use that pattern for `data-dfn-for` and `data-export`)

Reply:
Ah, I simply didn't know what `:is()` was for :)
I shortened the selectors accordingly. They start to look like regular expressions, yummy!